### DISCUSSION TITLE
Allow new line in invoice metadata details

### DISCUSSION THREAD

**Discussion by learntheropes** [btcpayserver/btcpayserver#4941] [Ideas / Feature Requests]
New lines in metadata properties are escaped when shown on the invoice details page (but they are kept if you retrieve the invoice with greenfield api).

Please consider adding  `white-space: pre-wrap;` to the css of the relevant div, so the field is shown correctly.

My use case is that one of the metadata field is a PGP public key. If the merchant copies the key as shown now is unsalable, since the new line is a requirement in the PGP public key.

As of now, I'm forced to build a merchant dashboard just for that, when she/he could have all the needed info under the btcpay dashboard. 

My use case is the PGP public key where the new line is a must, but the same can apply to any metadata property provided using a textarea (multiline) input field, where the merchant should see it properly formatted, as typed by the buyer.

I'm not a security expert, but I don't think this have any implication.

**Comment by dennisreimann** [comment]
With dashboard you mean the invoice detail page? Asking because we do not show the meta data on the store's dashboard. Maybe to clarify, can you post a redacted screenshot of where this would need fixing?

**Reply by learntheropes** [reply]
On the invoice page on btcpay I see the metadata under "Additional information".

See the screenshot: 

![screenshot](https://nostr.build/i/nostr.build_1fe7c735ffcc0cb6a0818d81feeded678a3de7419af3e3428a330656fd2990b2.jpg)

I edited the issue, sorry for that.

**Reply by dennisreimann** [reply]
Up to now we don't have a way to accept multiline input values via the forms. The input type text strips the newlines and replaces them with whitespace, so there wasn't even a way to solve this correctly.

I've opened #4942 to properly handle this.

**Reply by learntheropes** [reply]
Thank you a lot.

Just to clarify: I don't use your form because it is limited. I use my form with calendar, multiselect form validation etc and I use the metadata property to save the info with the Greenfield api.

**Reply by dennisreimann** [reply]
Ah ok, it makes sense that your values can actually contain newlines then. However, the PR addresses that issue on our end as well and adds proper display for both cases :)

**Reply by dennisreimann** [reply]
This will be released as part of 1.9.3 — also see #4955.
